utils.rb

```diff
@@ -0,0 +1,82 @@
+def create_person(option)
+  age = 0
+  loop do
+    print('Age: ')
+    age = gets.chomp.to_i
+    break if age.positive?
+  end
+  print 'Name: '
+  name = gets.chomp
+  case option
+  when 1
+    permission = ''
+    loop do
+      print 'Has parent permission? [Y/N]: '
+      permission = gets.chomp
+      break if %w[y n].include? permission.downcase
+    end
+    permission.downcase == 'y' ? Student.new(age, nil, name, true) : Student.new(age, nil, name, false)
+  else
+    print 'Specialization: '
+    specialization = gets.chomp
+    Teacher.new(age, specialization, name)
+  end
+  puts 'Person created successfully'
+  sleep 2
+end
+
+def create_book
+  print 'Title: '
+  title = gets.chomp
+  print 'Author: '
+  author = gets.chomp
+  Book.new(title, author)
+  puts 'Book created successfully'
+  sleep 2
+end
+
+def create_rental
+  all_books = Book::ALL_BOOKS.all_books
+  nbr_books = all_books.count
+  all_person = Person::ALL_PERSONS.all_persons
+  nbr_persons = all_person.size
+  book = -1
+  person = -1
+  loop do
+    puts 'Select a book from the following list by number'
+    all_books.each_with_index do |bk, s|
+      puts "#{s}) Title: #{bk.title}, Author: #{bk.author}"
+    end
+    book = gets.chomp.to_i
+    break if (0...nbr_books).include? book
+  end
+  loop do
+    puts 'Select a person from the following list by number (not id)'
+    all_person.each_with_index do |per, s|
+      puts "#{s}) [#{per.class.name}] Name: #{per.name}, ID: #{per.id}, Age: #{per.age}"
+    end
+    person = gets.chomp.to_i
+    break if (0...nbr_persons).include? person
+  end
+  date = ''
+  loop do
+    print 'Date: '
+    date = gets.chomp
+    break if date.match(%r{\d{4}/\d{1,2}/\d{1,2}$})
+  end
+  Rental.new(date, all_person[person], all_books[book])
+  puts 'Rental created successfully'
+  sleep 2
+end
```

Comment:
- [ ] The date for the rental is looping and does not accept the input you have incorrect syntax

![Capture d’écran 2022-01-31 160322](https://user-images.githubusercontent.com/61562626/151818104-410e3aac-1e30-4519-95ce-cdc99d2d4733.png)

Reply:
Hi @dasileker 
Thank yo for your review.
My code make control to date input data, so it will loop aver and over until he gets input data which can meet this format '----/--/--'. You can look at this piece of code which handle this feature.
https://github.com/Taher-web-dev/OOP-school-library/blob/refactor_code/utils.rb#L65